main.go

```diff
@@ -99,15 +92,19 @@ func main() {
 		TimeEncoder: zapcore.RFC3339TimeEncoder,
 	}
 	opts.BindFlags(flag.CommandLine)
-	flag.Parse()
 
-	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+	logger := zap.New(zap.UseFlagOptions(&opts))
+	ctrl.SetLogger(logger)
+	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
```

Comment:
`opts.BindFlags` uses `flags` while `capifags.AddDiagnosticsOptions` uses `pflags`. Here, we add the default `flag.FlagSet` to default `pflag.FlagSet`